### PR TITLE
WIP: Optmimize Multiversion storage to avoid duplicates

### PR DESCRIPTION
**Pull request by dickolsson**

### CODE REVIEW

File: src/Tests/MigrationTest.php [hunk 1]
```diff
@@ -41,7 +41,7 @@ class MigrationTest extends WebTestBase {
    * @var array
    */
   protected $entityTypes = [
-    'entity_test' => [],
+//    'entity_test' => [],
```
Comment (timmillwood) [inline src/Tests/MigrationTest.php]: If this is not needed now, we should remove it.

File: src/Entity/Query/QueryTrait.php [hunk 2]
```diff
@@ -44,6 +49,22 @@ public function isNotDeleted() {
...
+   */
+  public function loadFromCurrentWorkspace() {
+    $this->currentWorkspace = TRUE;
+    return $this;
+  }
+
+  /**
+   * @see \Drupal\multiversion\Entity\Query\QueryInterface::loadFromAllWorkspaces()
+   */
+  public function loadFromAllWorkspaces() {
```
Comment (dickolsson) [inline src/Entity/Query/QueryTrait.php]: Let's rename this to `loadFromAnyWorkspace()`, I think that's more clear.

File: src/Entity/Storage/ContentEntityStorageTrait.php [hunk 3]
```diff
@@ -79,6 +104,16 @@ public function loadUnchanged($id) {
...
   public function loadMultiple(array $ids = NULL) {
     $this->isDeleted = FALSE;
+    $this->currentWorkspace = TRUE;
+    return parent::loadMultiple($ids);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function loadFromAllWorkspaces(array $ids = NULL) {
```
Comment (dickolsson) [inline src/Entity/Storage/ContentEntityStorageTrait.php]: Same here, let's rename to `loadFromAnyWorkspace()`.

File: src/Entity/Storage/ContentEntityStorageTrait.php [hunk 4]
```diff
@@ -95,55 +130,79 @@ public function loadDeleted($id) {
...
   public function loadMultipleDeleted(array $ids = NULL) {
     $this->isDeleted = TRUE;
+    $this->currentWorkspace = TRUE;
     return parent::loadMultiple($ids);
   }
 
   /**
    * {@inheritdoc}
    */
+  public function loadAllByProperties(array $values = array()) {
```
Comment (dickolsson) [inline src/Entity/Storage/ContentEntityStorageTrait.php]: I think the name is a bit confusing here. Are we really loading "all"? Perhaps `loadByPropertiesFromAnyWorkspace()` would be clearer?

**Pull request merged**